### DISCUSSION TITLE
tuning solver_ref and solver_imp params to avoid mujoco exceptions

### DISCUSSION THREAD

**Discussion by shivanimall** [google-deepmind/mujoco#63] [Asking for Help] [answered]
While using hollow_cylinder objects I have to fine-tune the solver_ref and solver_imp parameter to avoid the below exception: 

```mujoco_py.builder.MujocoException: Got MuJoCo Warning: Nan, Inf or huge value in QACC at DOF 9. The simulation is unstable.```

This happens when a random action is applied to the object or the object accidentally falls down. I am facing a lot of these exceptions which is impacting my downstream tasks.

Do you have any suggestion/documentation/algorithmic approach on how to tune these based on objects' attributes?
Applying brute-force fine-tuning approach is quite intensive in my case.

Many thanks, please let me know if you need more details.

**Comment by yuvaltassa** [comment, marked as answer]
Hi,
Sorry for the delay. MuJoCo's current defaults are quite soft, which can lead to a lot of penetration and subsequent large forces and divergence. I t will probably help to stiffen everything globally by reducing solref[0] for all geoms. Another possibility is to reduce the time-step.
```
<default>
  <geom solref=".004 1"/>
</default>
```

**Comment by shivanimall** [comment]
Hi @yuvaltassa thanks for the response.  I tried reducing solref for all geoms to [0.004, 1.], plus trying control time-step range from 1/20 to 1/100 but facing the same issue. To note, I have composite objects with different sizes in different episode runs. (I create procedurally generated objects in robosuite, as [here](https://robosuite.ai/docs/modules/objects.html#creating-a-procedurally-generated-object))

Would I need to further fine-tune this param based on object size? Many thanks for your help! (also, posted as follow-up on #98 )

**Comment by yuvaltassa** [comment]
These timesteps are still quite large. Did you try anything in the vicinity of 1ms?

Another thing to try is to switch the integrator from [Euler to RK4](https://mujoco.readthedocs.io/en/latest/computation.html#geintegration).

If neither of these work for you, please post an example model where the failure is reproducible and we'll continue to investigate.
